### ISSUE TITLE
pdfmake.createPdf is not a function

### ISSUE THREAD

**Issue by cyberco** [bpampuch/pdfmake#522]
I'm bundling my app with webpack but when running:

```
pdfmake = require 'pdfmake'
pdfmake.createPdf(doc_definition).download('test.pdf')
```

I get the error:

```
pdfmake.createPdf is not a function
```

I've tried [this suggestion](http://stackoverflow.com/questions/27421528/pdfmake-use-in-an-application-with-webpack) of adding the script loader to Webpack and stating:

```
pdfmake = require 'script!pdfmake'
```

but that didn't help. Any suggestions where the error might be?

**Comment by circy** [comment]
Can you tell me please times the complete file on Plunker give?
Then I can look at the time, for me it work.

**Comment by cyberco** [comment]
I will (later tonight), but for now I can tell you (as I've said [here](http://stackoverflow.com/questions/35494376/pdfmake-pdfmake-createpdf-is-not-a-function)) that the only thing I'm doing is:

**Comment by circy** [comment]
Oh sry pdfmake = require 'pdfmake' you can use only Server side.
For me in Webpack works with the script loader.

Downloade the files from: https://github.com/bpampuch/pdfmake/tree/master/build
and loade the files with the script loader. npm install script-loader --save-dev

**Comment by circy** [comment]
Just give me humble if it works, if not, I published a tutorial to. I was there brought to mind! : D

**Comment by cyberco** [comment]
`pdfmake = require 'pdfmake'` surely works in the browser using webpack (mind you, it's LiveScript, in plain JS it would be `pdfmake = require('pdfmake');`.

I've tried downloading `pdfmake.js` and loading it with script-loader, but that doesn't work. Lots of errors. Besides, packaging the bundles that way takes a lot of time (but that's a different topic).

Part of the problem could be, as stated [here](https://github.com/bpampuch/pdfmake/issues/150), that npm installs the serverside version of pdfmake, with a package.json that has the following line:

```
"main": "src/printer.js",
```

I'm sorry @circy , but your English is a bit hard to follow. If German is easier for you, just add that to your replies as well (I can read German perfectly fine). Thanks for your help so far.

**Comment by ardean** [comment]
@cyberco a build with the webpack umd-wrapper should fix your problem #526 (no script-loader)

**Comment by danishin** [comment]
@cyberco I've forked this [here](https://github.com/danishin/pdfmake). It supports distributing via npm as well as webpack.

**Comment by spsaucier** [comment]
`npm install -S danishin/pdfmake`
`var pdfmake = require("../../../node_modules/pdfmake/build/pdfmake.min.js");`

**Comment by Brieg** [comment]
+1

Add to package.json in to dependencies `"pdfmake": "*"`
`npm install`
`var pdfmake = require('pdfmake');`

In code:
`var docDefinition = { content: 'This is an sample PDF printed with pdfMake' };`
`pdfmake.createPdf(docDefinition).download('test.pdf')`

Got:
`pdfmake.createPdf is not a function`

**Comment by circy** [comment]
@Brieg Does this happen in the browser or in Node Js?

**Comment by Wabbala** [comment]
It is in the browser​. Before I moved pdfmake to the server side, I also
experienced this issue. The web-app I build is based on requirejs/backbone/
and the solution for me was to put pdfmake in the index.html (or any other
main file you prefer).
Good luck!
